lib/ruby_lsp/requests/on_type_formatting.rb

```diff
@@ -84,8 +84,30 @@ def handle_statement_end
 
         indents = " " * @indentation
 
-        add_edit_with_text(" \n#{indents}end")
-        move_cursor_to(@position[:line], @indentation + 2)
+        if @previous_line.include?("\n")
+          # If the previous line has a line break, then it means there's content after the line break that triggered
+          # this completion. For these cases, we want to add the `end` after the content and move the cursor back to the
+          # keyword that triggered the completion
+
+          line = @position[:line]
+
+          # If there are enough lines in the document, we want to add the `end` token on the line below the extra
+          # content. Otherwise, we want to insert and extra line break ourselves
+          correction = if T.must(@document.source[@line_end..-1]).count("\n") >= 2
```

Comment:
How does this play out in the case of: 

```ruby
if something
immediate_content
```

Will it result in: 
``` ruby 
if 
  something 
endimmediate_content
```

Reply:
Good catch. Yeah, indeed we end up in a scenario like that. I'll think about how we can avoid it.

Reply:
I frankly haven't found an easy way to handle this. How do you feel about merging this one and trying to address this other case later?

To fix this, we would need to perform some lookahead operations to figure out what exists in the next line.

Reply:
I'm happy with that